c7n/resources/ecs.py

```diff
@@ -525,3 +538,145 @@ def process_instance(self, client, cluster, instance):
         except (client.exceptions.NoUpdateAvailableException,
                 client.exceptions.UpdateInProgressException):
             return
+
+
+@ECSCluster.action_registry.register('tag')
+@TaskDefinition.action_registry.register('tag')
+@Service.action_registry.register('tag')
+@Task.action_registry.register('tag')
+@ContainerInstance.action_registry.register('tag')
+class TagEcsResource(Tag):
+    """Action to create tag(s) on an ECS resource
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    Requires arns in new format for tasks, services, and container-instances.
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: tag-ecs-service
+                resource: ecs-service
+                filters:
+                  - "tag:target-tag": absent
+                  - type: taggable
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-value
+    """
+    permissions = ('ecs:TagResource',)
+
+    def process_resource_set(self, client, resources, tags):
+        mid = self.manager.resource_type.id
+        for r in resources:
+            if not ecs_taggable(self.manager.resource_type, r):
+                raise PolicyExecutionError("New format ecs arn required: " + r[mid])
+
+        tags = [{'key': t['Key'], 'value': t['Value']} for t in tags]
+        for r in resources:
```

Comment:
You're already iterating over the resources in line 574, please move the tagging under there so there isn't a second iteration over the list

Reply:
My thought here was to check all resources affected by the policy before tagging any. If someone writes a policy to tag a large amount of resources, they might get half way through the resource list and then fail

Reply:
looking at this again, I think there may be a larger issue here-- existing resources will not be moved over the new arn format in this case there may be resources that never get migrated over, we shouldn't fail the entire policy run because of that (since they're untaggable). I would just log the number of untagged resources at the end and tagging the rest

Reply:
ok we're now skipping over resources that cant be tagged, tagging the rest, and then logging a message that includes a count of those failures. Also added the same logic to untag